Automattic/jetpack
# Write Brief: Load unconfident words from backend and update caching

Fixes ##39899

## Proposed changes:

* Renames files to kebab-case
* Moves the dictionary fetching mechanism to a separate file
* Checks a file's `last-modified` header value for cache invalidation, making a HEAD request once when opening the editor
* Moves the unconfident words list to the backend

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch, open a new browser tab with the developer console open, make sure to have `debug` enabled with `localStorage.setItem( 'debug', '*' );`.
* Open the block editor
* Check the debug messages. For the spelling mistakes and unconfident words features you should see those messages:
  *  `Fetching feature data for type: <feature>. language: en`
  * `Loaded data from server <feature> en`
* Check that the localStorage entries for both features are populated with `requestTime` and `data`.
* Check that the previous localStorage entry for the spelling mistakes (`jetpack-ai-breve-spelling-context-en`) is removed
* Check the network tab. You should see two requests for `en.json`, one for each feature.
* Reload the page
* You should see those messages:
  *  `Fetching feature data for type: <feature>. language: en`
  *  `Data not modified <feature> en`
  * `Loaded data from local storage <feature> en`
* Check that those features are working as expected
* Sandbox `s0.wp.com`, `s1.wp.com` and `s2.wp.com`
* Change the content of one of those files
* Reload the page
* The cache should invalidate and the files should be requested again
